examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -279,55 +256,45 @@ export class TaskList extends DataObject implements ITaskList {
 		}
 
 		// TODO: Delete any items that are in the root but missing from the external data
-		const updateTaskPs = updatedExternalData.map(async ([id, { name, priority }]) => {
-			// Write external data into savedData map.
-			const currentTask = this.savedData.get<PersistedTask>(id);
-			let incomingNameDiffersFromSavedName = false;
-			let incomingPriorityDiffersFromSavedPriority = false;
-			// Create a new task because it doesn't exist already
-			if (currentTask === undefined) {
-				const savedNameString = SharedString.create(this.runtime);
-				const savedPriorityCell = SharedCell.create(this.runtime) as ISharedCell<number>;
-				const savedDataPT: PersistedTask = {
-					id,
-					name: savedNameString.handle as IFluidHandle<SharedString>,
-					priority: savedPriorityCell.handle as IFluidHandle<ISharedCell<number>>,
-				};
-				savedNameString.insertText(0, name);
-				savedPriorityCell.set(priority);
-				this.savedData.set(id, savedDataPT);
-			} else {
-				// Make changes to exisiting saved tasks
-				const [savedNameString, savedPriorityCell] = await Promise.all([
-					currentTask.name.get(),
-					currentTask.priority.get(),
-				]);
-				if (savedNameString.getText() !== name) {
-					incomingNameDiffersFromSavedName = true;
+		const updateTaskPs = incomingExternalData.map(
+			async ([id, { name: incomingName, priority: incomingPriority }]) => {
```

Comment:
It made sense to make this map of promises before when we had an `await` call inside, but right now this code is completely sync so I think it's clearer to just iterate normally over `incomingExternalData`.